src/lib/plot/axis/x-axis.js

```diff
@@ -20,39 +20,67 @@
 
 import React from 'react';
 
-import PureRenderComponent from '../pure-render-component';
+import PureRenderComponent from '../../pure-render-component';
 import Axis from './axis';
-import {getTicksTotalFromSize} from '../utils/axis-utils';
+import {ORIENTATION, getTicksTotalFromSize} from '../../utils/axis-utils';
 
-class XAxis extends PureRenderComponent {
+const {TOP, BOTTOM} = ORIENTATION;
 
-  static get requiresSVG() {
-    return true;
-  }
+const propTypes = {
+  ...Axis.propTypes,
+  orientation: React.PropTypes.oneOf([
+    TOP, BOTTOM
+  ])
+};
 
-  render() {
+const defaultProps = {
+  orientation: BOTTOM,
+  attr: 'x'
+};
+
+class XAxis extends PureRenderComponent {
+
+  _getDefaultAxisProps() {
```

Comment:
It seems like these computation steps should be performed at the very last minute in the render of the `Axis` as they're required computation for each `orientation` to work. That way, users can also use `Axis`, and `XAxis`, `YAxis` are just simple wrappers to apply a few defaults.